common/pool/goroutine_pool.go

```diff
@@ -0,0 +1,40 @@
+package pool
+
+type Pool struct {
+	size       int
+	taskChan   chan func()
+	cancelChan chan struct{}
+}
+
+func NewPool(size int) *Pool {
+	return &Pool{
+		size:     size,
+		taskChan: make(chan func()),
+	}
+}
+
+func (p *Pool) Start() {
```

Comment:
Please do not implement routine pool by ourselves when there is some good thirdpart lib.

Reply:
Replaced with `ants` library